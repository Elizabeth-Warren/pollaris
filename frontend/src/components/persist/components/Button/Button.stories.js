import React from 'react';
import styled from 'styled-components';
import { action } from '@storybook/addon-actions';
import { select, text } from '@storybook/addon-knobs';
import Button, { ButtonLevels, ButtonSizes } from './index';
import Chevron, { ChevronDirections } from '../Chevron';
import theme from '../../theme';

const getLevelOptions = () => select('Level', Object.values(ButtonLevels), Button.defaultProps.size);
const getSizeOptions = () => select('Size', Object.values(ButtonSizes), Button.defaultProps.size);
const handleClick = action('button clicked');

export default {
  title: 'Components/Button',
  component: Button,
  parameters: { jest: ['Button'] },
};

export const withKnobs = () => {
  const buttonText = text('Text', 'Click me');
  const level = getLevelOptions();
  const size = getSizeOptions();
  return (<Button level={level} size={size} onClick={handleClick}>{buttonText}</Button>);
};

export const asALink = () => {
  const level = getLevelOptions();
  const size = getSizeOptions();
  return (
    <Button
      level={level}
      size={size}
      onClick={handleClick}
      as="a"
      href="https://www.google.com"
      title="Opens in a new tab"
      target="_blank"
      rel="noreferrer"
    >
      Go to Google
    </Button>
  );
};

export const withChevron = () => {
  const buttonText = text('Text', 'Click me');
  const level = getLevelOptions();
  const size = getSizeOptions();
  const direction = select('Chevron', Object.values(ChevronDirections), Chevron.defaultProps.direction);
  return (
    <Button level={level} size={size} onClick={handleClick}>
      {buttonText}
      <Chevron direction={direction} style={{ marginLeft: '0.3em' }} />
    </Button>
  );
};

export const gallery = () => {
  const Wrapper = styled.div`
    display: grid;
    grid-gap: ${theme.spacing.sp3};
    grid-template-columns: repeat(auto-fill, minmax(400px, 1fr));
  `;
  const Title = styled.h2`
    font-size: ${theme.fontSize.lg};
    margin: 0;
  `;
  const Size = styled.h3`
    margin-bottom: ${theme.spacing.sp1};
    margin-top: ${theme.fontSize.sp2};
  `;
  const buttonText = text('Text', 'Click me');
  return (
    <Wrapper>
      {Object.values(ButtonLevels).map((level) => (
        <div key={level}>
          <Title>{level}</Title>
          {Object.values(ButtonSizes).map((size) => (
            <React.Fragment key={`${level}-${size}`}>
              <Size>{size}</Size>
              <Button onClick={handleClick} level={level} size={size}>{buttonText}</Button>
              {' '}
              <Button onClick={handleClick} level={level} size={size} disabled>{buttonText}</Button>
            </React.Fragment>
          ))}
        </div>
      ))}
    </Wrapper>
  );
};
