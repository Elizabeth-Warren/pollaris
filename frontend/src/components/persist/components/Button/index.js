import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../theme';

export const ButtonLevels = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
  QUATERNARY: 'quaternary',
  INVERTED: 'inverted',
  WHITE: 'white',
};

export const ButtonSizes = {
  SM: 'sm',
  MD: 'md',
  LG: 'lg',
};

const ButtonWrapper = styled.button`
  appearance: none;
  background: ${({ level }) => {
    switch (level) {
      case ButtonLevels.SECONDARY:
        return theme.colors.navy;
      case ButtonLevels.TERTIARY:
        return theme.colors.liberty;
      case ButtonLevels.QUATERNARY:
        return theme.colors.purple;
      case ButtonLevels.INVERTED:
        return 'transparent';
      case ButtonLevels.WHITE:
        return theme.colors.white;
      case ButtonLevels.PRIMARY:
      default:
        return theme.colors.red;
    }
  }};
  border: 4px solid ${({ level }) => {
    switch (level) {
      case ButtonLevels.SECONDARY:
      case ButtonLevels.INVERTED:
        return theme.colors.navy;
      case ButtonLevels.TERTIARY:
        return theme.colors.liberty;
      case ButtonLevels.QUATERNARY:
        return theme.colors.purple;
      case ButtonLevels.WHITE:
        return theme.colors.white;
      case ButtonLevels.PRIMARY:
      default:
        return theme.colors.red;
    }
  }};
  border-radius: 0;
  color: ${({ level }) => {
    switch (level) {
      case ButtonLevels.TERTIARY:
      case ButtonLevels.INVERTED:
      case ButtonLevels.WHITE:
        return theme.colors.navy;
      case ButtonLevels.PRIMARY:
      case ButtonLevels.SECONDARY:
      case ButtonLevels.QUATERNARY:
      default:
        return theme.colors.white;
    }
  }};
  cursor: pointer;
  display: inline-block;
  font-family: ${theme.fontFamily.compressed};
  font-size: ${({ size }) => {
    switch (size) {
      case ButtonSizes.SM:
        return theme.fontSize.lg;
      case ButtonSizes.MD:
      case ButtonSizes.LG:
      default:
        return theme.fontSize.xl;
    }
  }};
  line-height: ${theme.leading.none};
  padding: ${({ size }) => {
    switch (size) {
      case ButtonSizes.LG:
        return `${theme.spacing.sp0} ${theme.spacing.sp2} ${theme.spacing.sp1}`;
      case ButtonSizes.SM:
      case ButtonSizes.MD:
      default:
        return `${theme.spacing.sp0} ${theme.spacing.sp2}`;
    }
  }};
  text-decoration: none;
  text-transform: uppercase;
  transition-duration: 150ms;
  transition-property: background-color, border-color, color, opacity;
  transition-timing-function: ease-in;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    background-color: ${({ level }) => {
    switch (level) {
      case ButtonLevels.INVERTED:
        return theme.colors.navy;
      default:
        return 'transparent';
    }
  }};
    color: ${({ level }) => {
    switch (level) {
      case ButtonLevels.INVERTED:
      case ButtonLevels.WHITE:
        return theme.colors.white;
      case ButtonLevels.PRIMARY:
      case ButtonLevels.SECONDARY:
      case ButtonLevels.TERTIARY:
      case ButtonLevels.QUATERNARY:
      default:
        return theme.colors.navy;
    }
  }};
  }

  @media (min-width: ${theme.screens.md}) {
    font-size: ${({ size }) => size === ButtonSizes.LG && theme.fontSize['2xl']};
  }
`;

const Button = ({
  children, level, size, ...other
}) => (
  <ButtonWrapper level={level} size={size} {...other}>{children}</ButtonWrapper>
);

Button.propTypes = {
  /** Child text or elements */
  children: PropTypes.node.isRequired,
  /** Determines button color. One of 'primary', 'secondary', 'tertiary', 'inverted', or 'white'. */
  level: PropTypes.oneOf(Object.values(ButtonLevels)),
  /** Button size. one of 'sm', 'md', or 'lg. */
  size: PropTypes.oneOf(Object.values(ButtonSizes)),
};

Button.defaultProps = {
  level: ButtonLevels.PRIMARY,
  size: ButtonSizes.MD,
};

export default Button;
