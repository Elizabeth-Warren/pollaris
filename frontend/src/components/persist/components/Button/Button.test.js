import React from 'react';
import renderer from 'react-test-renderer';
import Button, { ButtonLevels, ButtonSizes } from './index';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <div>
        {Object.values(ButtonLevels).map((level) => (
          <div key={level}>
            {Object.values(ButtonSizes).map((size) => (
              <React.Fragment key={`${level}-${size}`}>
                <Button level={level} size={size}>Click me</Button>
                {' '}
                <Button level={level} size={size} disabled>Click me</Button>
              </React.Fragment>
            ))}
          </div>
        ))}
      </div>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
