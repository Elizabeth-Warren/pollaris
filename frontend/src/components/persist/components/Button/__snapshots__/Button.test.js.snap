// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #b61b28;
  border: 4px solid #b61b28;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 22px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c0:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #b61b28;
  border: 4px solid #b61b28;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c1:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c1:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #b61b28;
  border: 4px solid #b61b28;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c2:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c2:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #232444;
  border: 4px solid #232444;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 22px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c3:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c3:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c4 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #232444;
  border: 4px solid #232444;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c4:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c4:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #232444;
  border: 4px solid #232444;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c5:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c5:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c6 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #b7e4cf;
  border: 4px solid #b7e4cf;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 22px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c6:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c6:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c7 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #b7e4cf;
  border: 4px solid #b7e4cf;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c7:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c7:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c8 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #b7e4cf;
  border: 4px solid #b7e4cf;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c8:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c8:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c9 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #575993;
  border: 4px solid #575993;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 22px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c9:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c9:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c10 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #575993;
  border: 4px solid #575993;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c10:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c10:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c11 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #575993;
  border: 4px solid #575993;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c11:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c11:hover:not(:disabled) {
  background-color: transparent;
  color: #232444;
}

.c12 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 4px solid #232444;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 22px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c12:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c12:hover:not(:disabled) {
  background-color: #232444;
  color: #fff;
}

.c13 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 4px solid #232444;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c13:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c13:hover:not(:disabled) {
  background-color: #232444;
  color: #fff;
}

.c14 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 4px solid #232444;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c14:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c14:hover:not(:disabled) {
  background-color: #232444;
  color: #fff;
}

.c15 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #fff;
  border: 4px solid #fff;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 22px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c15:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c15:hover:not(:disabled) {
  background-color: transparent;
  color: #fff;
}

.c16 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #fff;
  border: 4px solid #fff;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c16:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c16:hover:not(:disabled) {
  background-color: transparent;
  color: #fff;
}

.c17 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #fff;
  border: 4px solid #fff;
  border-radius: 0;
  color: #232444;
  cursor: pointer;
  display: inline-block;
  font-family: 'Ringside Compressed A','Ringside Compressed B','Helvetica',sans-serif;
  font-size: 26px;
  line-height: 1;
  padding: 4px 16px 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,border-color,color,opacity;
  transition-property: background-color,border-color,color,opacity;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
}

.c17:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c17:hover:not(:disabled) {
  background-color: transparent;
  color: #fff;
}

@media (min-width:768px) {
  .c2 {
    font-size: 36px;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 36px;
  }
}

@media (min-width:768px) {
  .c8 {
    font-size: 36px;
  }
}

@media (min-width:768px) {
  .c11 {
    font-size: 36px;
  }
}

@media (min-width:768px) {
  .c14 {
    font-size: 36px;
  }
}

@media (min-width:768px) {
  .c17 {
    font-size: 36px;
  }
}

<div>
  <div>
    <button
      className="c0"
      size="sm"
    >
      Click me
    </button>
     
    <button
      className="c0"
      disabled={true}
      size="sm"
    >
      Click me
    </button>
    <button
      className="c1"
      size="md"
    >
      Click me
    </button>
     
    <button
      className="c1"
      disabled={true}
      size="md"
    >
      Click me
    </button>
    <button
      className="c2"
      size="lg"
    >
      Click me
    </button>
     
    <button
      className="c2"
      disabled={true}
      size="lg"
    >
      Click me
    </button>
  </div>
  <div>
    <button
      className="c3"
      size="sm"
    >
      Click me
    </button>
     
    <button
      className="c3"
      disabled={true}
      size="sm"
    >
      Click me
    </button>
    <button
      className="c4"
      size="md"
    >
      Click me
    </button>
     
    <button
      className="c4"
      disabled={true}
      size="md"
    >
      Click me
    </button>
    <button
      className="c5"
      size="lg"
    >
      Click me
    </button>
     
    <button
      className="c5"
      disabled={true}
      size="lg"
    >
      Click me
    </button>
  </div>
  <div>
    <button
      className="c6"
      size="sm"
    >
      Click me
    </button>
     
    <button
      className="c6"
      disabled={true}
      size="sm"
    >
      Click me
    </button>
    <button
      className="c7"
      size="md"
    >
      Click me
    </button>
     
    <button
      className="c7"
      disabled={true}
      size="md"
    >
      Click me
    </button>
    <button
      className="c8"
      size="lg"
    >
      Click me
    </button>
     
    <button
      className="c8"
      disabled={true}
      size="lg"
    >
      Click me
    </button>
  </div>
  <div>
    <button
      className="c9"
      size="sm"
    >
      Click me
    </button>
     
    <button
      className="c9"
      disabled={true}
      size="sm"
    >
      Click me
    </button>
    <button
      className="c10"
      size="md"
    >
      Click me
    </button>
     
    <button
      className="c10"
      disabled={true}
      size="md"
    >
      Click me
    </button>
    <button
      className="c11"
      size="lg"
    >
      Click me
    </button>
     
    <button
      className="c11"
      disabled={true}
      size="lg"
    >
      Click me
    </button>
  </div>
  <div>
    <button
      className="c12"
      size="sm"
    >
      Click me
    </button>
     
    <button
      className="c12"
      disabled={true}
      size="sm"
    >
      Click me
    </button>
    <button
      className="c13"
      size="md"
    >
      Click me
    </button>
     
    <button
      className="c13"
      disabled={true}
      size="md"
    >
      Click me
    </button>
    <button
      className="c14"
      size="lg"
    >
      Click me
    </button>
     
    <button
      className="c14"
      disabled={true}
      size="lg"
    >
      Click me
    </button>
  </div>
  <div>
    <button
      className="c15"
      size="sm"
    >
      Click me
    </button>
     
    <button
      className="c15"
      disabled={true}
      size="sm"
    >
      Click me
    </button>
    <button
      className="c16"
      size="md"
    >
      Click me
    </button>
     
    <button
      className="c16"
      disabled={true}
      size="md"
    >
      Click me
    </button>
    <button
      className="c17"
      size="lg"
    >
      Click me
    </button>
     
    <button
      className="c17"
      disabled={true}
      size="lg"
    >
      Click me
    </button>
  </div>
</div>
`;
